app/src/modules/extension/mv3RuleParser/utils.ts

```diff
@@ -60,15 +60,15 @@ const parseUrlParametersFromSource = (source: RulePairSource): ExtensionRuleCond
       case SourceOperator.MATCHES: {
         const { pattern, flags } = parseRegex(source.value);
         return {
-          regexFilter: `^https?://${pattern}(/|$)(.*)`,
+          regexFilter: `^https?://${pattern}[/?#$]?(.*)`,
```

Comment:
Wouldn't work if source condition is HOST+Regex `ample` and we try to open https://example.com

Reply:
RuleExecution shows it is executed but dnr is not applied

Reply:
How about this `^https?://[a-z0-9-]*${pattern}[a-z0-9:-]*.*`